routers/common/markup.go

```diff
@@ -66,6 +66,22 @@ func RenderMarkup(ctx *context.Base, repo *context.Repository, mode, text, urlPr
 		}
 	}
 
+	links := markup.Links{
+		AbsolutePrefix: true,
+		Base:           urlPrefix,
+	}
+
+	// Parse branch path and tree path, for correct media links.
+	urlElements := strings.Split(strings.TrimPrefix(urlPrefix, setting.AppURL), "/")
+	if len(urlElements) >= 5 && urlElements[2] == "src" {
```

Comment:
We should document why we expect the third element to be `src`
```suggestion
	// The expected route is '<user>/<repo>/src/[branch, commit or tag]/<identifier>'
	if len(urlElements) >= 5 && urlElements[2] == "src" {
```
and yes, this feels indeed really hacky.
I can think of a few of ways that might or might not work:
- user profiles
- viewing a markdown file in a commit
- viewing a markdown file in a tag
- viewing a directory with a README file

Reply:
Commit and tag should work.

The naming here is also a bit confusing because this `urlPrefix` is actually just the `Context` option for the API. And if in the future user profiles or directory READMEs were rendered with this they could construct a `Context` in the expected format for the API call.